src/com/xilinx/rapidwright/router/RouteNode.java

```diff
@@ -334,22 +335,43 @@ public RouteNode getBaseWire(){
 		return new RouteNode(n);
 	}
 	
-
+	public ArrayList<PIP> getPIPsForwardToSink(){
```

Comment:
This doesn't seem very NOC related?

Reply:
Most of this code was aimed at building out relocate-ability for the NOC.  I'm assuming some additional support methods were needed and I think that is why these are here.